mode2.lua

```diff
@@ -121,11 +137,11 @@ function courseplay:handle_mode2(vehicle, dt)
 			frontTractor = vehicle.cp.activeCombine.courseplayers[vehicle.cp.positionWithCombine - 1]
 			courseplay:debug(string.format('%s: activeCombine ~= nil, my position=%d, frontTractor (positionWithCombine %d) = %q', nameNum(vehicle), vehicle.cp.positionWithCombine, vehicle.cp.positionWithCombine - 1, nameNum(frontTractor)), 4);
 			--	courseplay:follow_tractor(vehicle, dt, tractor)
-			courseplay:setModeState(vehicle, 6);
+			courseplay:setModeState(vehicle, STATE_FOLLOW_TARGET_TRACTOR);
 			courseplay:unload_combine(vehicle, dt)
 		end
 	else -- NO active combine
-		if vehicle.cp.nextTargets ~= nil and #vehicle.cp.nextTargets > 0 and vehicle.cp.modeState == 5 and vehicle.cp.lastActiveCombine then
+		if vehicle.cp.nextTargets ~= nil and #vehicle.cp.nextTargets > 0 and vehicle.cp.modeState == STATE_FOLLOW_TARGET_WPS and vehicle.cp.lastActiveCombine then
 			courseplay:unload_combine(vehicle, dt)
 		else
 			-- STOP!!
```

Comment:
And then there's this mysterious mode 99 below in line 154. And that driveInDirection call. These seem to send the tractor back to the start waypoint when the trailer is full and this just drives through fruit. Not sure where to intercept this to create a path around the fruit.